ironfish/src/mining/pool.ts

```diff
@@ -421,26 +422,45 @@ export class MiningPool {
     this.webhooks.map((w) => w.poolStatus(status))
   }
 
-  async getStatus(): Promise<MiningPoolStatus> {
+  async getStatus(publicAddress?: string): Promise<MiningStatusMessage> {
     const [hashRate, sharesPending] = await Promise.all([
       this.estimateHashRate(),
       this.shares.sharesPendingPayout(),
     ])
 
-    return {
+    let minerCount = 0
+    this.stratum.clients.forEach((client) => {
+      if (client.subscribed) {
+        minerCount++
+      }
+    })
+
+    const status = {
       name: this.name,
       hashRate: hashRate,
-      miners: this.stratum.clients.size,
+      miners: minerCount,
       sharesPending: sharesPending,
       banCount: this.stratum.peers.banCount,
     }
-  }
-}
 
-export type MiningPoolStatus = {
-  name: string
-  hashRate: number
-  miners: number
-  sharesPending: number
-  banCount: number
+    if (publicAddress) {
+      let addressMiners = 0
+      this.stratum.clients.forEach((client) => {
```

Comment:
It's small, but do we need to create a closure here for each status call instead of just using a for loop and iterator from the map.values()? Normally I think it's what ever but there may be a lot of people connected that are using this.

Reply:
Good call, I think what I really wanted here was something like a Python generator expression, but I didn't think about closures.